server/controllers/schemaController.js

```diff
@@ -1,23 +1,74 @@
 const pool = require('./database');
 
 const schemaController = {
-  createSchema: (req, res, next) => {
-    // extract all the form inputs
-    const { schemaname } = req.body;
+  getSchema: (req, res, next) => {
+    // expecting to receive user_id and schema_id from req.body
+    const { user_id, schema_id } = req.body;
+    // query the table using user_id and schema_id
+    pool.query(
+      'SELECT * FROM Schemas WHERE user_id=$1 AND schema_id=$2',
+      [user_id, schema_id],
+      (err, result) => {
+        if (err) {
+          console.error(err);
+          return res.status(400).json({ error: 'error from getSchema' });
+        }
+        console.log('schemaController => getSchema', result.rows);
+        return res.status(200).json(result.rows);
+      }
+    );
+  },
+  updateSchema: (req, res, next) => {
+    // expecting to receive user_id and post_id and other fields that we want to update from req.body
+    const {
+      user_id,
+      schema_id,
+      schema_name,
+      key,
+      type,
+      options_check,
+      unique_check,
+      required_check
+    } = req.body;
 
-    // create the table
+    // query for the table
     pool.query(
-      // have to add more options
-      `CREATE TABLE IF NOT EXISTS ${schemaname}(key VARCHAR(50), type VARCHAR(50), options boolean, required boolean, default_value VARCHAR(50), user_id INT)`,
+      'UPDATE Schemas SET schema_name=$1 key=$2 type=$3 options_check=$4 unique_check=$5 required_check=$6 WHERE user_id=$7 AND schema_id=$8',
+      [
+        schema_name,
+        key,
+        type,
+        options_check,
```

Comment:
needed to do some trickery for getting boolean values out of this. for options, have to see if the prop exists at all.

Reply:
We can figure out how to best do that, otherwise the code looks good.